rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -113,6 +113,7 @@ class Publisher : public PublisherBase
     this->do_intra_process_publish(message_seq);
   }
 
+  [[deprecated("use publish(*msg)")]]
```

Comment:
An alternative suggestion would be to publish a unique ptr. Especially in the intra process case that would be better. Maybe mention that case?